features/step_definitions/logging.rb

```diff
@@ -1056,3 +1056,125 @@
+    })
+    unless @result[:success]
+        step %Q/I run the :go_to_kibana_management_page web action/
+        step %Q/I perform the :create_index_pattern_in_kibana web action with:/, table(%{
+            | index_pattern_name | "#{pattern_name}" |
+         })
+         raise "Failed to create index pattern #{pattern_name}" unless @result[:success]
+    end
+end
+
+Given /^cluster-admin create the #{QUOTED} pattern for the #{WORD} user$/ do | pattern_name, who |
+    user(word_to_num(who))
+    user_name=user.name
+    user_token=user.cached_tokens.first
+    step %Q/I switch to cluster admin pseudo user/
+    step %Q/I use the "openshift-logging" project/
+    
+    rest_cmd="curl -s --connect-timeout 10  -XPOST \"http://localhost:5601/api/saved_objects/index-pattern/#{pattern_name}\" -H \"kbn-xsrf: true\" -H \"x-forwarded-user: #{user_name}\" -H \"securitytenant: __user__\" -H \"Authorization: Bearer #{user_token}\" -H \"Content-Type: application/json\" -d \'{\"attributes\": {\"title\": \"#{pattern_name}\", \"timeFieldName\": \"@timestamp\"}}\'"
+
+    step %Q/a pod becomes ready with labels:/, table(%{
+        | component= kibana|
+    })
+    @result = pod.exec("bash", "-c", rest_cmd, as: admin, container: 'kibana')
+    raise "Failed to create index pattern via rest api" unless @result[:success]
+end
+
+Given /^I can display the(?: "(.+)")? pod logs of the(?: "(.+)")? project under the(?: "(.+)")? pattern in kibana$/ do | pod_name, project_name, pattern_name |
+    # I switch to cluster admin pseudo user
+    pod_name ||= "*"
+    project_name ||= "*"
+    pattern_name ||= "app"
+    #lucene_query_string = "kubernetes.namespace_name: #{project_name} and kubernetes.pod_name: #{pod_name} and @timestamp:[now-15m TO now]"
+    lucene_query_string = "kubernetes.namespace_name: #{project_name} and kubernetes.pod_name: #{pod_name}"
+    step %Q/I perform the :select_index_pattern_in_kibana web action with:/,table(%{
+             | index_pattern_name | #{pattern_name} |
+    })
+    raise("Can not select to pattern #{ pattern_name}!") unless @result[:success]
+
+    step %Q/I perform the :search_doc_in_kibana web action with:/,table(%{
+             | search_string | #{lucene_query_string} |
+    })
+    raise("Can not refresh the search #{lucene_query_string}!") unless @result[:success]
+
+    retries = 3
+    while true
+        step %Q/I run the :check_log_count web action/
+        if @result[:success]
+           break
+        elsif retries > 0
+           log("Can not find documents using clause #{lucene_query_string} under pattern  #{ pattern_name}! retry.....")
+        else
+           raise("Can not find documents using clause #{lucene_query_string} under pattern  #{ pattern_name}! after 3 retry,  abort")
+        end
+        step %Q/I run the :click_refresh_button_in_kibana action/
+        retries -= 1
+    end
+end
+
+Given /^I have clusterlogging with(?: (\d+))? persistent storage ES$/ do |es_num|
```

Comment:
@anpingli  why do you add this step here? The step `I create clusterlogging instance with` doesn't work for you?

Reply:
@QiaolingTang  In this step, it only create clusterlogging instance  when the ES doesn't have persistent volume.  it assume we can create clusterlogging manually before run case.

Reply:
What's the usage of ` create clusterlogging manually before run case` ? I think we're doing some duplicated things here.

Reply:
This step can provide the ability to run upgrade on a given cluster. For example:  Cluster logging with customized configuration.

Reply:
That is not duplicated with you step, that is an extend to your step.